### ISSUE TITLE
Cannot Save Plot with Bokeh 2.3

### ISSUE THREAD

**Issue by necromuralist** [holoviz/holoviews#4845]
##### from pypi
holoviews==1.14.1
hvplot==0.7.0
bokeh==2.3.0
pandas==1.2.2

##### OS
Ubuntu 20.10

##### Python
Python 3.8.6

##### Expected Behavior
`holoviews.save(plot, filename)` saves the bokeh plot.

##### Actual Behavior
Code raises an exception because bokeh now passes back a default `GlyphRenderer.selection_glyph` of "auto" instead of None and in `holoviews.plotting.bokeh.element.ElementPlot._update_glyph` this line:

```
dataspecs = glyph.dataspecs()
```

is being called when `glyph` is set to `"auto"`. Based on the git log it looks like the change might have been in the bokeh git commit: fbe1fa8b4df76ff6436312fb54097ce6b68edd89 (made on January 11).

where the `bokeh.models.renderers.GlyphRenderer.selection_glyph` was changed from an `Instance` class with a default of `None` to a `Either` class with a default of `"auto"`.

I'm not familiar with the code, so that's just a guess.. Reverting bokeh to 2.2.3 fixes it.

#### Complete, minimal, self-contained example code that reproduces the issue

```
import holoviews
import hvplot.pandas
import pandas

URL = "http://jse.amstat.org/v9n3/stanton.html#appendixa"

frames = pandas.read_html(URL, header=0)
data = frames[1]
data.columns = ["Parent", "Daughter", "Frequency"]
plot = data.hvplot(x="Parent", y="Daughter")
holoviews.save(plot, "test_plot")
```

#### Stack traceback and/or browser JavaScript console output

Traceback (most recent call last):
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/plotting/bokeh/element.py", line 1359, in _init_glyphs
    self._update_glyph(renderer, properties, mapping, glyph, source, source.data)
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/plotting/bokeh/element.py", line 1252, in _update_glyph
    dataspecs = glyph.dataspecs()
AttributeError: 'str' object has no attribute 'dataspecs'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 14, in <module>
    holoviews.save(plot, "test_plot")
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/util/__init__.py", line 819, in save
    return renderer_obj.save(obj, filename, fmt=fmt, resources=resources,
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/plotting/renderer.py", line 609, in save
    plot, fmt = self_or_cls._validate(obj, fmt)
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/plotting/renderer.py", line 314, in _validate
    plot = self.get_plot(obj, renderer=self, **kwargs)
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/plotting/bokeh/renderer.py", line 73, in get_plot
    plot = super(BokehRenderer, self_or_cls).get_plot(obj, doc, renderer, **kwargs)
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/plotting/renderer.py", line 243, in get_plot
    plot.update(init_key)
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/plotting/plot.py", line 982, in update
    return self.initialize_plot()
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/plotting/bokeh/element.py", line 1389, in initialize_plot
    self._init_glyphs(plot, element, ranges, source)
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/plotting/bokeh/element.py", line 1359, in _init_glyphs
    self._update_glyph(renderer, properties, mapping, glyph, source, source.data)
  File "/home/hades/.virtualenvs/Visions-Voices-Data/lib/python3.8/site-packages/holoviews/core/options.py", line 227, in __exit__
    raise AbbreviatedException(etype, value, traceback)
holoviews.core.options.AbbreviatedException: AttributeError: 'str' object has no attribute 'dataspecs'

**Comment by philippjfr** [comment]
A new holoviews release will be out soon, for now you can use a dev version with:

```
pip install panel holoviews --pre
```

or:

```
conda install -c pyviz/label/dev panel holoviews
```

**Comment by necromuralist** [comment]
Hello, thanks for the info. The dev version did fix it for me. But just in case someone else finds this I'll add that since I already had it installed I had to upgrade:

```
pip install panel holoviews --pre --upgrade
pip install bokeh --upgrade
```
Once again, thanks for the quick reply. If I had just posted the issue maybe I could have saved myself some time trying to figure out what was going on... but I suppose that would be a bad habit to pick up.
